// primeng/bundles/primeng-steps.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/router'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common'), require('@angular/router')) :
    typeof define === 'function' && define.amd ? define('primeng/steps', ['@angular/core','@angular/router','@angular/common','exports', '@angular/core', '@angular/common', '@angular/router'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.ng.core,global.ng.router,global.ng.common,(global.primeng = global.primeng || {}, global.primeng.steps = {}), global.ng.core, global.ng.common, global.ng.router));
}(this, (function (ɵngcc0,ɵngcc1,ɵngcc2,exports, core, common, router) { 
function Steps_li_2_a_2_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 11);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r1168 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r1168.label);
} }
function Steps_li_2_a_2_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 12);
} if (rf & 2) {
    var item_r1168 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵproperty("innerHTML", item_r1168.label, ɵngcc0.ɵɵsanitizeHtml);
} }
var _c0 = function () { return { exact: false }; };
function Steps_li_2_a_2_Template(rf, ctx) { if (rf & 1) {
    var _r1180 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "a", 7);
    ɵngcc0.ɵɵlistener("click", function Steps_li_2_a_2_Template_a_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r1180); var ctx_r1181 = ɵngcc0.ɵɵnextContext(); var item_r1168 = ctx_r1181.$implicit; var i_r1169 = ctx_r1181.index; var ctx_r1179 = ɵngcc0.ɵɵnextContext(); return ctx_r1179.itemClick($event, item_r1168, i_r1169); })("keydown.enter", function Steps_li_2_a_2_Template_a_keydown_enter_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r1180); var ctx_r1183 = ɵngcc0.ɵɵnextContext(); var item_r1168 = ctx_r1183.$implicit; var i_r1169 = ctx_r1183.index; var ctx_r1182 = ɵngcc0.ɵɵnextContext(); return ctx_r1182.itemClick($event, item_r1168, i_r1169); });
    ɵngcc0.ɵɵelementStart(1, "span", 8);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, Steps_li_2_a_2_span_3_Template, 2, 1, "span", 9);
    ɵngcc0.ɵɵtemplate(4, Steps_li_2_a_2_ng_template_4_Template, 1, 1, "ng-template", null, 10, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var _r1175 = ɵngcc0.ɵɵreference(5);
    var ctx_r1184 = ɵngcc0.ɵɵnextContext();
    var item_r1168 = ctx_r1184.$implicit;
    var i_r1169 = ctx_r1184.index;
    var ctx_r1171 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("routerLink", item_r1168.routerLink)("queryParams", item_r1168.queryParams)("routerLinkActive", "p-menuitem-link-active")("routerLinkActiveOptions", item_r1168.routerLinkActiveOptions || ɵngcc0.ɵɵpureFunction0(16, _c0))("fragment", item_r1168.fragment)("queryParamsHandling", item_r1168.queryParamsHandling)("preserveFragment", item_r1168.preserveFragment)("skipLocationChange", item_r1168.skipLocationChange)("replaceUrl", item_r1168.replaceUrl)("state", item_r1168.state);
    ɵngcc0.ɵɵattribute("target", item_r1168.target)("id", item_r1168.id)("tabindex", item_r1168.disabled || ctx_r1171.readonly ? null : item_r1168.tabindex ? item_r1168.tabindex : "0");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(i_r1169 + 1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", item_r1168.escape !== false)("ngIfElse", _r1175);
} }
function Steps_li_2_ng_template_3_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 11);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r1168 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r1168.label);
} }
function Steps_li_2_ng_template_3_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 12);
} if (rf & 2) {
    var item_r1168 = ɵngcc0.ɵɵnextContext(2).$implicit;
    ɵngcc0.ɵɵproperty("innerHTML", item_r1168.label, ɵngcc0.ɵɵsanitizeHtml);
} }
function Steps_li_2_ng_template_3_Template(rf, ctx) { if (rf & 1) {
    var _r1191 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "a", 13);
    ɵngcc0.ɵɵlistener("click", function Steps_li_2_ng_template_3_Template_a_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r1191); var ctx_r1192 = ɵngcc0.ɵɵnextContext(); var item_r1168 = ctx_r1192.$implicit; var i_r1169 = ctx_r1192.index; var ctx_r1190 = ɵngcc0.ɵɵnextContext(); return ctx_r1190.itemClick($event, item_r1168, i_r1169); })("keydown.enter", function Steps_li_2_ng_template_3_Template_a_keydown_enter_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r1191); var ctx_r1194 = ɵngcc0.ɵɵnextContext(); var item_r1168 = ctx_r1194.$implicit; var i_r1169 = ctx_r1194.index; var ctx_r1193 = ɵngcc0.ɵɵnextContext(); return ctx_r1193.itemClick($event, item_r1168, i_r1169); });
    ɵngcc0.ɵɵelementStart(1, "span", 8);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, Steps_li_2_ng_template_3_span_3_Template, 2, 1, "span", 9);
    ɵngcc0.ɵɵtemplate(4, Steps_li_2_ng_template_3_ng_template_4_Template, 1, 1, "ng-template", null, 14, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var _r1186 = ɵngcc0.ɵɵreference(5);
    var ctx_r1195 = ɵngcc0.ɵɵnextContext();
    var item_r1168 = ctx_r1195.$implicit;
    var i_r1169 = ctx_r1195.index;
    var ctx_r1173 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵattribute("href", item_r1168.url, ɵngcc0.ɵɵsanitizeUrl)("target", item_r1168.target)("id", item_r1168.id)("tabindex", item_r1168.disabled || i_r1169 !== ctx_r1173.activeIndex && ctx_r1173.readonly ? null : item_r1168.tabindex ? item_r1168.tabindex : "0");
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(i_r1169 + 1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", item_r1168.escape !== false)("ngIfElse", _r1186);
} }
var _c1 = function (a0, a1) { return { "p-highlight p-steps-current": a0, "p-disabled": a1 }; };
function Steps_li_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "li", 3, 4);
    ɵngcc0.ɵɵtemplate(2, Steps_li_2_a_2_Template, 6, 17, "a", 5);
    ɵngcc0.ɵɵtemplate(3, Steps_li_2_ng_template_3_Template, 6, 7, "ng-template", null, 6, ɵngcc0.ɵɵtemplateRefExtractor);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r1168 = ctx.$implicit;
    var i_r1169 = ctx.index;
    var _r1172 = ɵngcc0.ɵɵreference(4);
    var ctx_r1167 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵclassMap(item_r1168.styleClass);
    ɵngcc0.ɵɵproperty("ngStyle", item_r1168.style)("ngClass", ɵngcc0.ɵɵpureFunction2(8, _c1, ctx_r1167.isActive(item_r1168, i_r1169), item_r1168.disabled || ctx_r1167.readonly && !ctx_r1167.isActive(item_r1168, i_r1169)));
    ɵngcc0.ɵɵattribute("aria-selected", i_r1169 === ctx_r1167.activeIndex)("aria-expanded", i_r1169 === ctx_r1167.activeIndex);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1167.isClickableRouterLink(item_r1168))("ngIfElse", _r1172);
} }
var _c2 = function (a1) { return { "p-steps p-component": true, "p-readonly": a1 }; };
'use strict';

    var Steps = /** @class */ (function () {
        function Steps(router, route, cd) {
            this.router = router;
            this.route = route;
            this.cd = cd;
            this.activeIndex = 0;
            this.readonly = true;
            this.activeIndexChange = new core.EventEmitter();
        }
        Steps.prototype.ngOnInit = function () {
            var _this = this;
            this.subscription = this.router.events.subscribe(function () { return _this.cd.markForCheck(); });
        };
        Steps.prototype.itemClick = function (event, item, i) {
            if (this.readonly || item.disabled) {
                event.preventDefault();
                return;
            }
            this.activeIndexChange.emit(i);
            if (!item.url) {
                event.preventDefault();
            }
            if (item.command) {
                item.command({
                    originalEvent: event,
                    item: item,
                    index: i
                });
            }
        };
        Steps.prototype.isClickableRouterLink = function (item) {
            return item.routerLink && !this.readonly && !item.disabled;
        };
        Steps.prototype.isActive = function (item, index) {
            if (item.routerLink)
                return this.router.isActive(item.routerLink, false) || this.router.isActive(this.router.createUrlTree([item.routerLink], { relativeTo: this.route }).toString(), false);
            else
                return index === this.activeIndex;
        };
        Steps.prototype.ngOnDestroy = function () {
            this.subscription.unsubscribe();
        };
Steps.ɵfac = function Steps_Factory(t) { return new (t || Steps)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.Router), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.ActivatedRoute), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
Steps.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: Steps, selectors: [["p-steps"]], inputs: { activeIndex: "activeIndex", readonly: "readonly", model: "model", style: "style", styleClass: "styleClass" }, outputs: { activeIndexChange: "activeIndexChange" }, decls: 3, vars: 7, consts: [[3, "ngClass", "ngStyle"], ["role", "tablist"], ["class", "p-steps-item", "role", "tab", 3, "ngStyle", "class", "ngClass", 4, "ngFor", "ngForOf"], ["role", "tab", 1, "p-steps-item", 3, "ngStyle", "ngClass"], ["menuitem", ""], ["role", "presentation", "class", "p-menuitem-link", 3, "routerLink", "queryParams", "routerLinkActive", "routerLinkActiveOptions", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "click", "keydown.enter", 4, "ngIf", "ngIfElse"], ["elseBlock", ""], ["role", "presentation", 1, "p-menuitem-link", 3, "routerLink", "queryParams", "routerLinkActive", "routerLinkActiveOptions", "fragment", "queryParamsHandling", "preserveFragment", "skipLocationChange", "replaceUrl", "state", "click", "keydown.enter"], [1, "p-steps-number"], ["class", "p-steps-title", 4, "ngIf", "ngIfElse"], ["htmlLabel", ""], [1, "p-steps-title"], [1, "p-steps-title", 3, "innerHTML"], ["role", "presentation", 1, "p-menuitem-link", 3, "click", "keydown.enter"], ["htmlRouteLabel", ""]], template: function Steps_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "ul", 1);
        ɵngcc0.ɵɵtemplate(2, Steps_li_2_Template, 5, 11, "li", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵclassMap(ctx.styleClass);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(5, _c2, ctx.readonly))("ngStyle", ctx.style);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.model);
    } }, directives: [ɵngcc2.NgClass, ɵngcc2.NgStyle, ɵngcc2.NgForOf, ɵngcc2.NgIf, ɵngcc1.RouterLinkWithHref, ɵngcc1.RouterLinkActive], styles: [".p-steps{position:relative}.p-steps ul{display:flex;list-style-type:none;margin:0;padding:0}.p-steps-item{display:flex;flex:1 1 auto;justify-content:center;position:relative}.p-steps-item .p-menuitem-link{align-items:center;display:inline-flex;flex-direction:column;overflow:hidden;text-decoration:none}.p-steps.p-steps-readonly .p-steps-item{cursor:auto}.p-steps-item.p-steps-current .p-menuitem-link{cursor:default}.p-steps-title{white-space:nowrap}.p-steps-number{align-items:center;display:flex;justify-content:center}.p-steps-title{display:block}"], encapsulation: 2, changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(Steps, [{
        type: core.Component,
        args: [{
                selector: 'p-steps',
                template: "\n        <div [ngClass]=\"{'p-steps p-component':true,'p-readonly':readonly}\" [ngStyle]=\"style\" [class]=\"styleClass\">\n            <ul role=\"tablist\">\n                <li *ngFor=\"let item of model; let i = index\" class=\"p-steps-item\" #menuitem [ngStyle]=\"item.style\" [class]=\"item.styleClass\" role=\"tab\" [attr.aria-selected]=\"i === activeIndex\" [attr.aria-expanded]=\"i === activeIndex\"\n                    [ngClass]=\"{'p-highlight p-steps-current': isActive(item, i), 'p-disabled': item.disabled || (readonly && !isActive(item, i))}\">\n                    <a *ngIf=\"isClickableRouterLink(item); else elseBlock\" [routerLink]=\"item.routerLink\" [queryParams]=\"item.queryParams\" role=\"presentation\" [routerLinkActive]=\"'p-menuitem-link-active'\" [routerLinkActiveOptions]=\"item.routerLinkActiveOptions||{exact:false}\" class=\"p-menuitem-link\" \n                        (click)=\"itemClick($event, item, i)\" (keydown.enter)=\"itemClick($event, item, i)\" [attr.target]=\"item.target\" [attr.id]=\"item.id\" [attr.tabindex]=\"item.disabled || readonly ? null : (item.tabindex ? item.tabindex : '0')\"\n                        [fragment]=\"item.fragment\" [queryParamsHandling]=\"item.queryParamsHandling\" [preserveFragment]=\"item.preserveFragment\" [skipLocationChange]=\"item.skipLocationChange\" [replaceUrl]=\"item.replaceUrl\" [state]=\"item.state\">\n                        <span class=\"p-steps-number\">{{i + 1}}</span>\n                        <span class=\"p-steps-title\" *ngIf=\"item.escape !== false; else htmlLabel\">{{item.label}}</span>\n                        <ng-template #htmlLabel><span class=\"p-steps-title\" [innerHTML]=\"item.label\"></span></ng-template>\n                    </a>\n                    <ng-template #elseBlock>\n                        <a [attr.href]=\"item.url\" class=\"p-menuitem-link\" role=\"presentation\" (click)=\"itemClick($event, item, i)\" (keydown.enter)=\"itemClick($event, item, i)\" [attr.target]=\"item.target\" [attr.id]=\"item.id\" \n                            [attr.tabindex]=\"item.disabled||(i !== activeIndex && readonly) ? null : (item.tabindex ? item.tabindex : '0')\">\n                            <span class=\"p-steps-number\">{{i + 1}}</span>\n                            <span class=\"p-steps-title\" *ngIf=\"item.escape !== false; else htmlRouteLabel\">{{item.label}}</span>\n                            <ng-template #htmlRouteLabel><span class=\"p-steps-title\" [innerHTML]=\"item.label\"></span></ng-template>\n                        </a>\n                    </ng-template>\n                </li>\n            </ul>\n        </div>\n    ",
                changeDetection: core.ChangeDetectionStrategy.OnPush,
                encapsulation: core.ViewEncapsulation.None,
                styles: [".p-steps{position:relative}.p-steps ul{display:flex;list-style-type:none;margin:0;padding:0}.p-steps-item{display:flex;flex:1 1 auto;justify-content:center;position:relative}.p-steps-item .p-menuitem-link{align-items:center;display:inline-flex;flex-direction:column;overflow:hidden;text-decoration:none}.p-steps.p-steps-readonly .p-steps-item{cursor:auto}.p-steps-item.p-steps-current .p-menuitem-link{cursor:default}.p-steps-title{white-space:nowrap}.p-steps-number{align-items:center;display:flex;justify-content:center}.p-steps-title{display:block}"]
            }]
    }], function () { return [{ type: ɵngcc1.Router }, { type: ɵngcc1.ActivatedRoute }, { type: ɵngcc0.ChangeDetectorRef }]; }, { activeIndex: [{
            type: core.Input
        }], readonly: [{
            type: core.Input
        }], activeIndexChange: [{
            type: core.Output
        }], model: [{
            type: core.Input
        }], style: [{
            type: core.Input
        }], styleClass: [{
            type: core.Input
        }] }); })();
        return Steps;
    }());
    Steps.ctorParameters = function () { return [
        { type: router.Router },
        { type: router.ActivatedRoute },
        { type: core.ChangeDetectorRef }
    ]; };
    Steps.propDecorators = {
        activeIndex: [{ type: core.Input }],
        model: [{ type: core.Input }],
        readonly: [{ type: core.Input }],
        style: [{ type: core.Input }],
        styleClass: [{ type: core.Input }],
        activeIndexChange: [{ type: core.Output }]
    };
    var StepsModule = /** @class */ (function () {
        function StepsModule() {
        }
StepsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: StepsModule });
StepsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function StepsModule_Factory(t) { return new (t || StepsModule)(); }, imports: [[common.CommonModule, router.RouterModule], ɵngcc1.RouterModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(StepsModule, { declarations: [Steps], imports: [ɵngcc2.CommonModule, ɵngcc1.RouterModule], exports: [Steps, ɵngcc1.RouterModule] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(StepsModule, [{
        type: core.NgModule,
        args: [{
                imports: [common.CommonModule, router.RouterModule],
                exports: [Steps, router.RouterModule],
                declarations: [Steps]
            }]
    }], function () { return []; }, null); })();
        return StepsModule;
    }());

    /**
     * Generated bundle index. Do not edit.
     */

    exports.Steps = Steps;
    exports.StepsModule = StepsModule;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
